sql/bgw_scheduler.sql

```diff
@@ -21,13 +21,13 @@ INSERT INTO _timescaledb_config.bgw_job (id, application_name, job_type, schedul
 (1, 'Telemetry Reporter', 'telemetry_and_version_check_if_enabled', INTERVAL '24h', INTERVAL '100s', -1, INTERVAL '1h')
 ON CONFLICT (id) DO NOTHING;
 
-CREATE OR REPLACE FUNCTION add_drop_chunks_policy(
+CREATE OR REPLACE FUNCTION add_retention_policy(
```

Comment:
Maybe we should take the opportunity to also add a description of this function:

```suggestion
-- Add a retention policy to a hypertable or continuous aggregate.
-- The retention_window (typically an INTERVAL) determines the 
-- window beyond which data is dropped at the time
-- of execution of the policy (e.g., '1 week'). Note that the retention 
-- window will always align with chunk boundaries, thus the window 
-- might be larger than the given one, but never smaller. In other
-- words, some data beyond the retention window 
-- might be kept, but data within the window will never be deleted.
CREATE OR REPLACE FUNCTION add_retention_policy(
```


Looking at this signature, it is unclear what the time epoch is from here the retention window starts. I assume it is wall clock (system clock), but I am not sure that always fits the data. We might need to add a separate parameter that allows the user to specify a function that gives the epoch, typically defaulting to `now()`, similar to what I proposed for the `add_refresh_policy` function. This is obviously for another PR, however.

Reply:
I'm not sure I see the case where one wouldn't assume this was wall clock?

Reply:
It is probably not the common case, but if the data ingested is not aligned with the wall clock, then the retention policy won't work. Let's say, for instance, you are importing data in `TIMESTAMP WITHOUT TIMEZONE` and the database runs in another timezone than where the data originated.

Or, let's say your data uses some completely different notion of time, e.g., an integer that is an increasing counter, e.g., event number. The question is if the retention internally uses the `integer_now_function` in that case to get the current notion of time when the policy runs?